### DISCUSSION TITLE
What features do you need to go live?

### DISCUSSION THREAD

**Discussion by ElTimuro** [documenso/documenso#2] [General]
- What features do you need to go live tomorrow?
- What workflows do you need?
- Is the developer preview enough for you to go live, feature-wise?

**Comment by philippeauriach** [comment]
Hello 👋
Some features that would be mandatory in one of our case:
- A way to generate documents to be signed using an API (for integration with an existing internal tool or even zappier for example)
- A way to sign documents in a mobile app (might be achievable using webviews?)

Sorry if those are already developed / planned, I did not see mention of them :)

**Comment by ElTimuro** [comment]
Hey, thanks for the input :) API is not explicitly mentioned yet, but of course, planned. There will be a more detailed product roadmap soon, I'll be sure to include some specifics. Zapier is a great idea, if you have specific zaps in mind feel free to post them, I would add them to the backlog. Would you prefer Zapier or a generic REST API first?

Not sure what you mean by "in a mobile app". As part of a native android app workflow for example? Since the signing links require no login this should be straightforward with web views (disclaimer I'm not an app developer) or an in-app browser. If there is something else, let me know 👌

**Reply by philippeauriach** [reply]
Zapier was just an idea like this, we could imagine a contract to be sent after filling a typeform or something. I would definitely prefer a generic API !

If no login is needed then as long as the UI is responsive, a webview should be straightforward indeed (getting the signing link from the API mentioned before)

thanks a lot, will follow the project!

**Reply by ElTimuro** [reply]
Glad to hear it 🚀 Let me know if you have questions at any point.

**Reply by treed593** [reply]
I will chime in as another vote for a Generic REST API. We do most of our custom integrations without Zapier or similar tools and have been looking for something like Documenso

**Reply by bramkn** [reply]
As soon as a REST API is available, a node for n8n will follow soon. 
Already have some interest from some of my n8n clients for it. :-)

**Reply by fmerian** [reply]
@bramkn awesome!

for the record, there's an open issue for the n8n integration (#110).

**Comment by treed593** [comment]
It would be nice, if we could generate a "sign" link without the need to connect Documenso to an email provider. So each signer would still have a unique link, but we could send that manually in a separate process

**Reply by ElTimuro** [reply]
Good Idea. The would be the same as in the Emai IMO so it would just need to be copyable from the recipients from. Added an Issue here: https://github.com/documenso/documenso/issues/36

**Comment by yl-chow** [comment]
For my organisation's use case, the following are crucial:
1. Generate signing session via API (we prefer REST API but if that's not possible in the near future, could you consider making it available on Make (formerly Integromat)?)
2. From point number 1 above, generate signing link for each signer so that we can embed the links in our client-facing dashboards or send the links to clients via email/WhatsApp API
3. Signing sequence
4. Ability to control types of signature (e.g. turning off typed signatures; allowing only drawn signatures)
5. Audit trail
6. Ability to split audit trail from the signing document (in my market, some banks/government agencies are fussy about not having the audit trail attached with the main doc)

**Reply by fmerian** [reply]
@yl-chow hi YL, thanks for this detailed workflow!!

**Comment by gauravdalal-me** [comment]
@ElTimuro  Do you have API documentation which can be made available ? 
Also, how about Multi-Tenant Approach - E.g. We run a SAAS based model where we would like to offer this as a plugin to each Tenant to work independently using their own URL's , SMTP etc.

**Reply by fmerian** [reply]
@apptologyindia hello there! we don't have API documentation just yet as we're currently doing a rewrite (see roadmap [here](https://github.com/orgs/documenso/projects/1) and learn more about the rewrite [there](https://github.com/documenso/documenso/wiki/Why-we're-doing-a-rewrite)). it will definitely come up with the release tho. thanks!

**Comment by gillistephan** [comment]
For our use case in order to switch to a hosted Documenso solution immediately, the following would be required:

1. Generate a signing session via an API by providing a previously generated PDF document with defined fields (preferrably based on PDF form fields - which is pretty common in our industry - banks/insurance). Comparable to docusign [tabs](https://developers.docusign.com/docs/esign-rest-api/esign101/concepts/tabs/) concept, but less complicated.

2. Define a predefined signing sequence in API call from 1) - for our industry, order matters. 
3. Ability to generate a Signing-Link for each Signer

Nice to have: 
1. Embeddable React-Widget, which receives a Signing-Session or any other props, to be able to directly embedd the Signing in our product, to provide a seamless customer experience.
2. Ability to get notified (Webhooks) / request information about a document status (overall status, who has signed)

**Reply by fmerian** [reply]
@gillistephan thanks, Stephan! seems similar to @yl-chow's [use case](https://github.com/documenso/documenso/discussions/2#discussioncomment-6137684). also love the idea of embedding for less context-switching and notifications.

keep an eye on the [roadmap](https://github.com/orgs/documenso/projects/1?pane=issue&itemId=34655339)!

**Comment by Aj7Ay** [comment]
Hello,
Modify the existing signature field feature to accept and display user initials instead of the full name. When signing a document, users should input their initials instead of a complete name. Ensure that the system sends and stores the provided initials correctly for document signing purposes.

**Comment by tankerkiller125** [comment]
Generic OIDC SSO, or at least more options than just Google.

**Reply by jinnatar** [reply]
Have this same requirement, logged it as an FR after a bit of investigation: https://github.com/documenso/documenso/issues/1090

**Reply by tankerkiller125** [reply]
**Update:** The PR for this for self-hosting users has been merged, support for a wider scope on this will probably take more time as it's significantly more complex.

**Comment by PhilippvH1** [comment]
- API ✅
- LTV 
- Force users to sign using a hand/drawn signature every time 
- German tranlsation

**Comment by benjaminrancourt** [comment]
For my professional use, the possibility of adding people's initials on each page is missing. Yousign has created a [good explanatory document on initials](https://yousign.com/fr-fr/blog/parapher-document-valeur-juridique) and their legal value (in French though).

With this feature, Documenso would definitely be a DocuSign alternative for small businesses! 😉

**Comment by Phoenix-Ignited-Tech** [comment]
OIDC for integration with SSO solutions such as Keycloak etc.

**Reply by tankerkiller125** [reply]
This feature was added not that long ago (like a week maybe) for self-hosted installations. It allows a single OIDC provider of the self-hosters choosing via the environment variables config.

**Reply by Phoenix-Ignited-Tech** [reply]
Cool, I noticed the variables in the .env.example file shortly after I posted, also side note, is there an admin backend for self-hosters? I saw a discussion elsewhere regarding not being able to log into the admin side, I've scoured the .env.example file and tried to find any info on how to log in as an admin user but couldn't find anything.

**Reply by ephraimduncan** [reply]
There is an admin backend but you'll need a user with ADMIN role.

**Reply by tankerkiller125** [reply]
@ephraimduncan I'm guessing we would need to do this manually in the database?

**Reply by Phoenix-Ignited-Tech** [reply]
I found in this file: https://github.com/documenso/documenso/blob/main/packages/prisma/seed/initial-seed.ts an admin user with email of admin@documenso.com and a password of password (This logs you into the admin dashboard).  I've been trying to figure out what the password is for the documenso postgres db, so I can login with DBeaver and manually create an admin user, no luck yet.

**Reply by ephraimduncan** [reply]
@tankerkiller125 currently yes

@Phoenix-Ignited-Tech How are you self hosting it?

**Reply by Phoenix-Ignited-Tech** [reply]
I've only tested it locally, I followed the local development guide: https://github.com/documenso/documenso/tree/main#local-development

**Comment by Novapixel1010** [comment]
It would be nice to change the `signature disclosure` maybe be able to provide your own custom page. So something like just changing the URL would be good. I provide example here #1812 but no maintainer as commented on it. At the very least be able to change the support email so people can reach out. The current default is `support@documenso.com` which means people would be reaching out to documenso.
